### PR TITLE
use non-immediate dispatcher in camera component

### PR DESCRIPTION
**Pull request by Zayankovsky**
Alternative for #6442.

### CODE REVIEW

File: libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/camera/CameraComponent.kt [hunk 1]
```diff
@@ -219,7 +220,8 @@ internal class CameraComponent constructor(
                 viewportDataSource.evaluate()
             }
         }
-        coroutineScope.launch {
+        // using immediate dispatcher causes nested store updates
```
Comment (dzinad) [inline libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/camera/CameraComponent.kt:223]: That's a temporary solution, right? Could you add a TODO with a ticket?

**Review by dzinad — APPROVED** [review]

**Review by abhishek1508 — APPROVED** [review]

**Review by kmadsen — APPROVED** [review]

File: libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/camera/CameraComponent.kt (hunk 1 above)
Comment (kmadsen) [reply libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/camera/CameraComponent.kt:223]: could also consider this the follow up https://github.com/mapbox/mapbox-navigation-android/pull/6442

File: libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/camera/CameraComponent.kt [hunk 2]
```diff
@@ -219,7 +220,8 @@ internal class CameraComponent constructor(
...
+        coroutineScope.launch(Dispatchers.Main) {
```
Comment (tomaszrybakiewicz) [inline libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/camera/CameraComponent.kt:224]: All coroutines that call `store.dispatch` must be updated.

File: libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/camera/CameraComponent.kt (hunk 1 above)
Comment (Zayankovsky) [reply libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/camera/CameraComponent.kt:224]: Yeah, I hope we come up with something different. But not everyone agrees here.
So no TODO for now. There's another PR open already, so we won't forget about this.

**Comment by Zayankovsky** [comment]
@mapbox/navigation-android `checkNotificationContent` internal instrumentation test is failing consistently.

**Pull request merged**